File: core/src/com/qtech/bubbleblaster/common/screen/ScreenType.java
package com.qtech.bubbleblaster.common.screen;

import com.qtech.bubbleblaster.common.RegistryEntry;
import com.qtech.bubbleblaster.common.scene.Scene;

import java.util.Objects;
import java.util.function.Function;

@Deprecated
public class ScreenType<T extends Screen> extends RegistryEntry {
    // Fields
    private final Function<Scene, T> screenFactory;

    public ScreenType(Function<Scene, T> screenFactory) {
        // EffectInstance class field.
        this.screenFactory = screenFactory;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ScreenType<?> that = (ScreenType<?>) o;
        return Objects.equals(getRegistryName(), that.getRegistryName());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getRegistryName());
    }

    public T getScreen(Scene scene) {
        return screenFactory.apply(scene);
    }
}

File: src/main/kotlin/qtech/bubbles/common/screen/ScreenType.kt
@file:Suppress("DEPRECATION")

package qtech.bubbles.common.screen

import qtech.bubbles.common.RegistryEntry
import qtech.bubbles.common.scene.Scene
import java.util.*
import java.util.function.Function

@Deprecated("")
class ScreenType<T : Screen?>     // EffectInstance class field.
    (  // Fields
    private val screenFactory: Function<Scene, T>
) : RegistryEntry() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ScreenType<*>
        return registryName == that.registryName
    }

    override fun hashCode(): Int {
        return Objects.hash(registryName)
    }

    fun getScreen(scene: Scene): T {
        return screenFactory.apply(scene)
    }
}